webapp/src/Controller/API/ContestController.php

```diff
@@ -363,75 +363,74 @@ public function changeTimesAction(Request $request, string $cid): Response
             // We allow the "force" parameter to override this.
             if (!$request->request->getBoolean('force') &&
                 $contest->getStarttime() != null &&
                 $contest->getStarttime() < $now + 30) {
-                $response = new JsonResponse('Current contest already started or about to start.',
+                return new JsonResponse('Current contest already started or about to start.',
                     Response::HTTP_FORBIDDEN);
-            } elseif ($request->request->get('start_time') === null) {
-                $this->em->persist($contest);
+            }
+
+            if ($request->request->get('start_time') === null) {
                 $contest->setStarttimeEnabled(false);
                 $response = new Response('', Response::HTTP_NO_CONTENT);
                 $this->em->flush();
                 $changed = true;
             } else {
                 $date = date_create($request->request->get('start_time'));
                 if ($date === false) {
-                    $response = new JsonResponse('Invalid "start_time" in request.', Response::HTTP_BAD_REQUEST);
-                } else {
-                    $new_start_time = $date->getTimestamp();
-                    if (!$request->request->getBoolean('force') && $new_start_time < $now + 30) {
-                        $response = new JsonResponse('New start_time not far enough in the future.',
-                            Response::HTTP_FORBIDDEN);
-                    } else {
-                        $this->em->persist($contest);
-                        $newStartTimeString = date('Y-m-d H:i:s e', $new_start_time);
-                        $contest->setStarttimeEnabled(true);
-                        $contest->setStarttime($new_start_time);
-                        $contest->setStarttimeString($newStartTimeString);
-                        $response = new Response('', Response::HTTP_NO_CONTENT);
-                        $this->em->flush();
-                        $changed = true;
-                    }
+                    return new JsonResponse('Invalid "start_time" in request.', Response::HTTP_BAD_REQUEST);
+                }
+
+                $new_start_time = $date->getTimestamp();
+                if (!$request->request->getBoolean('force') && $new_start_time < $now + 30) {
+                    return new JsonResponse('New start_time not far enough in the future.',
+                        Response::HTTP_FORBIDDEN);
                 }
+                $newStartTimeString = date('Y-m-d H:i:s e', $new_start_time);
+                $contest->setStarttimeEnabled(true);
+                $contest->setStarttime($new_start_time);
+                $contest->setStarttimeString($newStartTimeString);
+                $response = new Response('', Response::HTTP_NO_CONTENT);
+                $this->em->flush();
+                $changed = true;
             }
         } else { // $request->request->has('scoreboard_thaw_time')
```

Comment:
Nitpick: I think this makes the control flow also easier to read. And due to the checks above this is equivalent.
```suggestion
        }
        if ($request->request->has('scoreboard_thaw_time')) {
```